Unable to start LSP on MacOS with ts plugin

### Describe the bug
When activating after initialising a gauge-ts project, the extensions was unable to start after repeatedly crashing.

side note: 
- initialising gauge-java and gauge-js seems to work fine, but referencing gave an error `failed to parse request json: cannot unmarshal array into Go value of type string`


### What command(s) did you run when you found the bug?
Activating the extension, selecting the template for gauge-ts

### Output, stack trace or logs related to the bug

lsp.log

        14-01-2021 18:14:31.669 [Gauge] [DEBUG] Attempting to connect to grpc server at port: 57622
        14-01-2021 18:14:31.672 [Gauge] [DEBUG] Successfully made the connection with runner with port: 57622
        14-01-2021 18:14:31.672 [Gauge] [INFO] LangServer: reading on stdin, writing on stdout
        14-01-2021 18:14:31.884 [Gauge] [DEBUG] Runner with PID:61704 has exited
        14-01-2021 18:14:31.885 [Gauge] [INFO] Connection closed
        14-01-2021 18:14:31.885 [Gauge] [ERROR] Unable to log error 'Connection closed' to LSP: jsonrpc2: connection is closed

### Versions

0.0.19

#### Gauge (Output of `gauge -v`)

        Gauge version: 1.1.7
        Plugins
        -------
        html-report (4.0.12)
        java (0.7.13)
        js (2.3.15)
        screenshot (0.0.1)
        ts (0.1.0)

#### Node.js/Java/Python/.Net/Ruby version

        node --version
        v14.9.0

#### Operating System information

MacOS Big Sur
Version 11.1


#### IDE information

Visual Studio Code Version: 1.52.1

--- a/.github/workflows/golangci-lint.yml
+++ b/.github/workflows/golangci-lint.yml
@@ -17,7 +17,7 @@ jobs:
         uses: golangci/golangci-lint-action@v2
         with:
           # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
-          version: v1.29
+          version: v1.39
 
           # Optional: working directory, useful for monorepos
           # working-directory: somedir

--- a/api/lang/diagnostics.go
+++ b/api/lang/diagnostics.go
@@ -127,7 +127,7 @@ func validateConcepts(diagnostics map[lsp.DocumentURI][]lsp.Diagnostic) (*gauge.
 			return nil, fmt.Errorf("unable to read file %s", err)
 		}
 		cpts, pRes := new(parser.ConceptParser).Parse(content, conceptFile)
-		pErrs, err := parser.AddConcept(cpts, conceptFile, conceptDictionary)
+		pErrs, err := parser.AddConcept(cpts, conceptFile, conceptDictionary) // nolint
 		if err != nil {
 			return nil, err
 		}

--- a/api/lang/references.go
+++ b/api/lang/references.go
@@ -17,11 +17,11 @@ import (
 )
 
 func stepReferences(req *jsonrpc2.Request) (interface{}, error) {
-	var params string
+	var params []string
 	if err := json.Unmarshal(*req.Params, &params); err != nil {
 		return nil, fmt.Errorf("failed to parse request %v", err)
 	}
-	return getLocationFor(params)
+	return getLocationFor(params[0])
 }
 
 func stepValueAt(req *jsonrpc2.Request) (interface{}, error) {

--- a/api/lang/references_test.go
+++ b/api/lang/references_test.go
@@ -33,7 +33,7 @@ Scenario Heading
 	openFilesCache = &files{cache: make(map[lsp.DocumentURI][]string)}
 	openFilesCache.add(uri, specText)
 
-	b, _ := json.Marshal("Say {} to {}")
+	b, _ := json.Marshal([]string{"Say {} to {}"})
 	params := json.RawMessage(b)
 	want := []lsp.Location{
 		{URI: uri, Range: lsp.Range{

--- a/cmd/run.go
+++ b/cmd/run.go
@@ -11,6 +11,7 @@ import (
 	"os"
 	"strconv"
 	"strings"
+	gauge "github.com/getgauge/gauge/gauge"
 	"github.com/getgauge/gauge/config"
 	"github.com/getgauge/gauge/env"
 	"github.com/getgauge/gauge/execution"

--- a/cmd/run.go
+++ b/cmd/run.go
@@ -10,11 +10,7 @@ import (
 	"fmt"
 	"os"
 	"strconv"
-
-	"github.com/getgauge/gauge/gauge"
-
 	"strings"
-
 	"github.com/getgauge/gauge/config"
 	"github.com/getgauge/gauge/env"
 	"github.com/getgauge/gauge/execution"

--- a/execution/merge.go
+++ b/execution/merge.go
@@ -81,7 +81,9 @@ func mergeResults(results []*result.SpecResult) *result.SpecResult {
 		if res.GetFailed() {
 			specResult.IsFailed = true
 		}
-		var tableRows []*m.ProtoTableRow
+
+		var tableRows []*m.ProtoTableRow // nolint
+
 		for _, item := range res.ProtoSpec.Items {
 			switch item.ItemType {
 			case m.ProtoItem_Scenario:

--- a/execution/result/specResult.go
+++ b/execution/result/specResult.go
@@ -55,7 +55,7 @@ func (specResult *SpecResult) AddTableDrivenScenarioResult(r *ScenarioResult, t 
 		specResult.ScenarioFailedCount++
 	}
 	specResult.AddExecTime(r.ExecTime())
-	pItem := &gauge_messages.ProtoItem{
+	pItem := &gauge_messages.ProtoItem{ // nolint
 		ItemType: gauge_messages.ProtoItem_TableDrivenScenario,
 		TableDrivenScenario: &gauge_messages.ProtoTableDrivenScenario{
 			Scenario:              r.Item().(*gauge_messages.ProtoScenario),
@@ -88,7 +88,7 @@ func (specResult *SpecResult) AddTableRelatedScenarioResult(scenarioResults [][]
 				TableRowIndex:    int32(index),
 				ScenarioTableRow: eachRow[scenarioIndex].(*ScenarioResult).ScenarioDataTableRow,
 			}
-			protoItem := &gauge_messages.ProtoItem{ItemType: gauge_messages.ProtoItem_TableDrivenScenario, TableDrivenScenario: protoTableDrivenScenario}
+			protoItem := &gauge_messages.ProtoItem{ItemType: gauge_messages.ProtoItem_TableDrivenScenario, TableDrivenScenario: protoTableDrivenScenario} // nolint
 			specResult.ProtoSpec.Items = append(specResult.ProtoSpec.Items, protoItem)
 		}
 		if scenarioFailed {

--- a/gauge/protoConverters.go
+++ b/gauge/protoConverters.go
@@ -154,7 +154,7 @@ func ConvertToProtoTable(table *Table) *gauge_messages.ProtoTable {
 	}
 	protoTableParam := &gauge_messages.ProtoTable{Rows: make([]*gauge_messages.ProtoTableRow, 0)}
 	protoTableParam.Headers = &gauge_messages.ProtoTableRow{Cells: table.Headers}
-	for _, row := range table.Rows() {
+	for _, row := range table.Rows() { // nolint
 		protoTableParam.Rows = append(protoTableParam.Rows, &gauge_messages.ProtoTableRow{Cells: row})
 	}
 	return protoTableParam

--- a/logger/logger_test.go
+++ b/logger/logger_test.go
@@ -17,7 +17,7 @@ import (
 	"github.com/getgauge/gauge/config"
 	"github.com/getgauge/gauge/plugin/pluginInfo"
 	"github.com/getgauge/gauge/version"
-	"github.com/op/go-logging"
+	logging "github.com/op/go-logging"
 )
 
 func TestGetLoggerShouldGetTheLoggerForGivenModule(t *testing.T) {

--- a/parser/dataTableSpecs.go
+++ b/parser/dataTableSpecs.go
@@ -24,7 +24,7 @@ func GetSpecsForDataTableRows(s []*gauge.Specification, errMap *gauge.BuildError
 				if len(tableRelatedScenarios) > 0 {
 					s := createSpecsForTableRows(spec, tableRelatedScenarios, errMap)
 					s[0].Scenarios = append(s[0].Scenarios, nonTableRelatedScenarios...)
-					for _, scn := range nonTableRelatedScenarios {
+					for _, scn := range nonTableRelatedScenarios { // nolint
 						s[0].Items = append(s[0].Items, scn)
 					}
 					specs = append(specs, s...)

--- a/runner/grpcRunner.go
+++ b/runner/grpcRunner.go
@@ -20,7 +20,6 @@ import (
 	"github.com/getgauge/gauge/config"
 	"github.com/getgauge/gauge/logger"
 	"github.com/getgauge/gauge/manifest"
-	"google.golang.org/genproto/googleapis/rpc/errdetails"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"

--- a/runner/grpcRunner.go
+++ b/runner/grpcRunner.go
@@ -20,6 +20,7 @@ import (
 	"github.com/getgauge/gauge/config"
 	"github.com/getgauge/gauge/logger"
 	"github.com/getgauge/gauge/manifest"
+	errdetails "google.golang.org/genproto/googleapis/rpc/errdetails"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"

--- a/validation/validate.go
+++ b/validation/validate.go
@@ -219,7 +219,7 @@ func fillScenarioErrors(scenario *gauge.Scenario, errMap *gauge.BuildErrors, ste
 		if step.IsConcept {
 			fillScenarioErrors(scenario, errMap, step.ConceptSteps)
 		}
-		if err, ok := errMap.StepErrs[step]; ok {
+		if err, ok := errMap.StepErrs[step]; ok { // nolint
 			errMap.ScenarioErrs[scenario] = append(errMap.ScenarioErrs[scenario], err)
 		}
 	}
@@ -230,7 +230,7 @@ func fillSpecErrors(spec *gauge.Specification, errMap *gauge.BuildErrors, steps 
 		if context.IsConcept {
 			fillSpecErrors(spec, errMap, context.ConceptSteps)
 		}
-		if err, ok := errMap.StepErrs[context]; ok {
+		if err, ok := errMap.StepErrs[context]; ok { // nolint
 			errMap.SpecErrs[spec] = append(errMap.SpecErrs[spec], err)
 			for _, scenario := range spec.Scenarios {
 				if _, ok := errMap.ScenarioErrs[scenario]; !ok {

--- a/validation/validation_test.go
+++ b/validation/validation_test.go
@@ -41,7 +41,7 @@ Scenario 2
 `
 	p := new(parser.SpecParser)
 	spec, _, _ := p.Parse(specText, gauge.NewConceptDictionary(), "")
-	err := gauge_messages.StepValidateResponse_STEP_IMPLEMENTATION_NOT_FOUND
+	err := gauge_messages.StepValidateResponse_STEP_IMPLEMENTATION_NOT_FOUND // nolint
 	errs := validationErrors{spec: []error{
 		NewStepValidationError(spec.Scenarios[0].Steps[0], "", "", &err, ""),
 		NewStepValidationError(spec.Scenarios[1].Steps[0], "", "", &err, ""),
@@ -67,7 +67,7 @@ Scenario 2
 `
 	p := new(parser.SpecParser)
 	spec, _, _ := p.Parse(specText, gauge.NewConceptDictionary(), "")
-	err := gauge_messages.StepValidateResponse_STEP_IMPLEMENTATION_NOT_FOUND
+	err := gauge_messages.StepValidateResponse_STEP_IMPLEMENTATION_NOT_FOUND // nolint
 
 	errs := validationErrors{spec: []error{
 		NewStepValidationError(spec.Scenarios[0].Steps[0], "", "", &err, ""),

--- a/version/version.go
+++ b/version/version.go
@@ -14,7 +14,7 @@ import (
 )
 
 // CurrentGaugeVersion represents the current version of Gauge
-var CurrentGaugeVersion = &Version{1, 1, 7}
+var CurrentGaugeVersion = &Version{1, 1, 8}
 
 // BuildMetadata represents build information of current release (e.g, nightly build information)
 var BuildMetadata = ""

